radix-engine/tests/common_transactions.rs

```diff
@@ -0,0 +1,338 @@
+            ),
+            account_component_address =
+                bech32_encoder.encode_component_address_to_string(&account_component_address)
+        );
+        (manifest, Vec::new())
+    });
+}
+
+/// An example manifest for creating a new fungible resource with an initial supply
+#[test]
+fn creating_a_fungible_resource_with_initial_supply_succeeds() {
+    test_manifest(|account_component_address, bech32_encoder| {
+        let initial_supply = Decimal::from("10000000");
+
+        let manifest = format!(
+            include_str!(
+                "../../transaction/examples/resources/creation/fungible/with_initial_supply.rtm"
+            ),
+            initial_supply = initial_supply,
+            account_component_address =
+                bech32_encoder.encode_component_address_to_string(&account_component_address)
+        );
+        (manifest, Vec::new())
+    });
+}
+
+/// An example manifest for creating a new non-fungible resource with no supply
+#[test]
+fn creating_a_non_fungible_resource_with_no_initial_supply_succeeds() {
+    test_manifest(|account_component_address, bech32_encoder| {
+        let manifest = format!(
+            include_str!(
+                "../../transaction/examples/resources/creation/non_fungible/no_initial_supply.rtm"
+            ),
+            account_component_address =
+                bech32_encoder.encode_component_address_to_string(&account_component_address)
+        );
+        (manifest, Vec::new())
+    });
+}
+
+/// An example manifest for creating a new non-fungible resource with an initial supply
+#[test]
+fn creating_a_non_fungible_resource_with_initial_supply_succeeds() {
+    test_manifest(|account_component_address, bech32_encoder| {
+        let manifest = format!(
+            include_str!("../../transaction/examples/resources/creation/non_fungible/with_initial_supply.rtm"),
+            account_component_address =
+                bech32_encoder.encode_component_address_to_string(&account_component_address)
+        );
+        (manifest, Vec::new())
+    });
+}
+
+/// A sample manifest that publishes a package.
+#[test]
+fn publish_package_with_owner_succeeds() {
+    test_manifest_with_owner_badge(
+        |account_component_address, owner_badge_non_fungible_address, bech32_encoder| {
+            let owner_badge_resource_address = owner_badge_non_fungible_address.resource_address();
+            let owner_badge_non_fungible_id = if let NonFungibleId::U32(non_fungible_id) =
+                owner_badge_non_fungible_address.non_fungible_id()
+            {
+                *non_fungible_id
+            } else {
+                panic!("expected a u32 non-fungible-id");
+            };
+
+            // TODO: Update the complex.abi and complex.code files that are used for testing.
+            // Using the WASM and ABI from the account blueprint here as they are up to date. The
+            // complex.code and complex.abi files from the transaction crate are not.
```

Comment:
👍

Do you think it might be worth separating the current contents of `transaction/examples` to eg `transaction/test-cases` or something? So that `transaction/examples` is full of only easy to use/follow manifests? :D

Reply:
Done. All of the manifests and blobs that used to be in the root of the examples directory have now been moved to `examples/test-cases` so that this is the one directory that people do not look at :D